### DISCUSSION TITLE
Initialize a load weights from a checkpoint into a `LightningModule` w/o loading in training state

### DISCUSSION THREAD

**Discussion by EthanMarx** [Lightning-AI/pytorch-lightning#20688] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I am trying to initialize a training run using a `LightningCLI`, where I want my `LightningModule` to _only load in the weights_ from a lightning checkpoint. 

I don't want the saved training state (e.g. scheduler / optimizer state) to be initialized from the checkpoint. It's my understanding that using the `--ckpt_path` flag will load in scheduler / optimizer state. 

How can I initialize a training run from pre-trained weights w/o loading in this additional state?

**Comment by mauvilsa** [comment]
I suppose you would need to add an optional parameter to your module and implement this logic. I don't know if there is any other way to do that.

**Reply by EthanMarx** [reply]
Thanks! Thats what I ended up doing. Just curious if there was a native lightning argument
